### ISSUE TITLE
Try fails to catch NullPointerException when using java converters on null values

### ISSUE THREAD

**Issue by ettoretorti** [scala/bug#10638]
The following code throws a NullPointerException instead of returning a Failure wrapping it.
```scala
import scala.collection.JavaConverters._
import scala.util.Try

Try((null : java.util.List[String]).asScala)
```
Is this intended behaviour?

Scala and java versions:
```sh
$ scala -version
Scala code runner version 2.11.8 -- Copyright 2002-2016, LAMP/EPFL
$ java -version
java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)
```

**Comment by SethTisue** [comment]
not a bug; `.asScala` succeeds, but then the `NullPointerException` occurs afterwards, when the REPL tries to print the result

in general, it's expected behavior that implicit conversions succeed even on `null`, unless there is particular code in the body of the implicit conversion that dereferences the value being converted

in general, "is this intended behavior?" questions are better on users.scala-lang.org (for beginners) or contributors.scala-lang.org (for more experienced users); come to scala/bug once you're reasonably certain you have a bug

**Comment by martijnhoekstra** [comment]
underlying issue is #9113 and fixed in 2.12

**Comment by SethTisue** [comment]
> in general, it's expected behavior that implicit conversions succeed even on null

I should add that it's certainly appropriate to add explicit null checks at Scala/Java boundaries, as was done for #9113
